### PR TITLE
Milestone 1: Plain Javascript with objects

### PR DESCRIPTION
**Pull request by alvarorf**
In this milestone, the following features are added:
- Create a collection that keeps a list of books.
- Create a function to add a new book to the collection, with title and author.
- Create a function to remove a book from the collection.
- Display all books saved in the collection in the top part of the page.
- It makes sure that when a user clicks the "Add" button: i.  A new book is added to the collection. ii.  The new book is displayed in the page.
- It makes sure that when a user clicks the "Remove" button:
    The correct book is removed from the collection.
    The correct book dissapears from the page.

- Data is preserved in the browser's memory by using localStorage.

Thank you for your comments.

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -0,0 +1,47 @@
+if (!localStorage.getItem('books')) {
+  localStorage.setItem('books', JSON.stringify([]));
+}
+
+const retrieveBooks = () => JSON.parse(localStorage.getItem('books'));
```
Comment (mateo951) [inline index.js:5]: Fantastic work 💪 You are using an array of objects for keeping the list of books 📝

File: index.js [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+};
+
+const removeBook = (id) => {
+  const books = retrieveBooks();
+  const updatedBooks = books.map((book) => {
+    if (book.id === id) { book.deleted = true; }
+    return book;
+  });
+  updateBooks(updatedBooks);
+};
```
Comment (mateo951) [inline index.js:45]: Add and remove book functions are working as a charm 🔥😎🔥

**Review by mateo951 — CHANGES_REQUESTED** [review]
### Hi Team, 

📝👌 **Your project highlights** 🔥😎 
- Github Flow, and PR as expected ✔
- JS Best Practices ✔
- The functionality of the app works perfectly 🙌

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)_

File: index.html [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+          <button type="button" onclick="addBook()">Add Book</button>
+        </li>
+      </ul>
+    </form>
+  </section>
+
+
+
+  <script src="index.js"></script>
+</body>
```
[the remark this answers is not in the document]
Comment (alvarorf) [reply index.html:38]: Thank you!

File: index.js [hunk 4]
```diff
@@ -0,0 +1,47 @@
+if (!localStorage.getItem('books')) {
+  localStorage.setItem('books', JSON.stringify([]));
+}
+
+const retrieveBooks = () => JSON.parse(localStorage.getItem('books'));
+
+const renderBooks = () => {
```
[the remark this answers is not in the document]
Comment (alvarorf) [reply index.js:7]: Thanks!

**Pull request merged**